src/v3/src/transformer/selectAuthenticator/transformSelectAuthenticatorEnroll.ts

```diff
@@ -64,23 +94,65 @@ export const transformSelectAuthenticatorEnroll: IdxStepTransformer = ({
       content: loc('oie.select.authenticators.enroll.title', 'login'),
     },
   };
-  const informationalText: DescriptionElement = {
+  const description: DescriptionElement = {
     type: 'Description',
     contentType: 'subtitle',
     options: getContentDescrAndParams(brandName),
   };
-  const description: HeadingElement = {
+
+  const headingRequiredNow: HeadingElement = {
+    type: 'Heading',
+    options: {
+      content: loc('oie.setup.required.now', 'login'),
+      level: 3,
+      visualLevel: 6,
+      dataSe: 'authenticator-list-title',
+    },
+  };
+
+  const headingRequiredSoon: HeadingElement = {
+    type: 'Heading',
+    options: {
+      content: loc('oie.setup.required.soon', 'login'),
+      level: 3,
+      visualLevel: 6,
+      dataSe: 'authenticator-list-title',
+    },
+  };
+
+  const headingNoGracePeriod: HeadingElement = {
     type: 'Heading',
     options: {
       content: skipStep ? loc('oie.setup.optional', 'login') : loc('oie.setup.required', 'login'),
       level: 3,
       visualLevel: 6,
     },
   };
-  const authenticatorListElement: AuthenticatorButtonListElement = {
-    type: 'AuthenticatorButtonList',
-    options: { buttons: authenticatorButtons, dataSe: 'authenticator-enroll-list' },
+
+  const descriptionGracePeriod: DescriptionElement = {
+    type: 'Description',
+    contentType: 'subtitle',
+    options: {
+      content: loc('oie.setup.required.soon.description', 'login'),
+    },
   };
+
+  const authenticatorListElementWithGracePeriod: AuthenticatorButtonListElement[] = [];
+  if (authenticatorButtonsWithGracePeriod.length) {
+    authenticatorListElementWithGracePeriod.push({
+      type: 'AuthenticatorButtonList',
+      options: { buttons: authenticatorButtonsWithGracePeriod, dataSe: 'authenticator-enroll-list-grace-period' },
+    });
+  }
+
+  const authenticatorListElementDueNow: AuthenticatorButtonListElement[] = [];
+  if (authenticatorButtonsDueNow.length) {
+    authenticatorListElementDueNow.push({
+      type: 'AuthenticatorButtonList',
+      options: { buttons: authenticatorButtonsDueNow, dataSe: 'authenticator-enroll-list' },
+    });
+  }
+
   const skipButton: ButtonElement = {
     type: 'Button',
     label: loc('oie.optional.authenticator.button.title', 'login'),
```

Comment:
@jakebacher-okta were you waiting to change this to "Remind me later" in the case when authenticators have grace period + skip ?